test/m4/configure.m4

```diff
@@ -7,30 +7,6 @@ AC_CONFIG_FILES([test/Makefile
 		 test/miscellaneous/Makefile
 		 test/performance/Makefile
 		 test/validation/Makefile
-		 test/validation/api/atomic/Makefile
```

Comment:
I choose more structured tests vs build time. Do not see good reason to put all files to one folder.

Reply:
@muvarov to speed up parallel build. Right now building tests involves changing CWD into and out of per-test directory. This can not be parallelized. `make -j 100` suffers.